WordPress/gutenberg
# Tabs: Make sure individual `Tab`s are linked to the correct `TabPanel`s

## What?
Fixes a bug that prevented `Tabs` from explicitly linking `Tab` sub-components to their respective `TabPanel` counterparts.

## Why?
`Ariakit.TabPanel` has an internal `tabId` prop that I appear to have overlooked when first implementing `Tabs`. This is the prop responsible for declaring which `tab` element a `tabpanel` should be controlled by. Without it, `Ariakit` falls back to linking the sub-components based on the order in which they're rendered. That's usually okay but definitely not always.

I actually remember including this prop initially when starting work on `Tabs`, my best guess is that I mistook it for a typo at some point and removed it.

## How?
I've reinstated the `tabId` prop that's passed to `Ariakit` internally, and added a note to clarify the difference between `id` (the DOM element's id) and `tabId` in this instance.

This does mean that `Ariakit` and `Tabs` each have a `tabId` prop that serve slightly different purposes, but they're closely enough related that I think it's okay.

I've also added a new unit test for this to prevent any future regressions.

## Testing Instructions
1. Confirm that the new unit test passes (aka, all tests should pass 😉)
2. Launch Storybook, and open the Tabs default story
3. Edit the default story by mixing up the order of the `Tabs.TabPanel` sub-components so they're in a different order than the corresponding `Tabs.Tab` components.
4. Confirm that the tabs still render the correct content, regardless of the order.
